pkg/registry/resource/resourceslice/strategy.go

```diff
@@ -210,6 +211,19 @@ func dropDisabledDRAPartitionableDevicesFields(newSlice, oldSlice *resource.Reso
 	}
 }
 
+func dropDisabledDRADeviceBindingConditionsFields(newSlice, oldSlice *resource.ResourceSlice) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.DRADeviceBindingConditions) && utilfeature.DefaultFeatureGate.Enabled(features.DRAResourceClaimDeviceStatus) {
```

Comment:
Don't drop the fields if the old slice already had them. Otherwise adding a label to an existing slice would wipe out the fields.

I'm torn about checking both features here. On the one hand, we don't want to allow setting fields that the cluster is not configured to support, so we have to check both. On the other hand the fields are said to be feature-gated only by DRADeviceBindingConditions.

I'm leaning towards updating the field comments to:
```
	// This is an alpha field and requires enabling the DRADeviceBindingConditions and DRAResourceClaimDeviceStatus
	// feature gates.
	//
	// +optional
	// +featureGate=DRADeviceBindingConditions,DRAResourceClaimDeviceStatus
```

But I need to ask around regarding this, it is unusual.

Reply:
For now, I will add a process to check the contents of oldslice.

Should I add comments? Is it okay to just provide instructions in the documentation?

Reply:
Jordan confirmed on Slack that the comment that I had proposed above is correct. Please add it.